util/hash_test.cc

```diff
@@ -584,6 +585,65 @@ TEST(MathTest, CodingGeneric) {
   EXPECT_EQ(std::string("_12"), std::string(out));
 }
 
+TEST(UniquenessTest, Approaches) {
+  ROCKSDB_NAMESPACE::Random64 r(getpid());
+
+  constexpr uint64_t kMaxIter = 10000;
+  constexpr size_t kSize = size_t{1} << 20;
+  constexpr size_t kMask = kSize - 1;
+
+  enum Approach {
+    kAddition,
+    kXor,
+    kRandom,
+  };
+
+  std::vector<bool> v;
+  for (auto approach : {kAddition, kXor, kRandom}) {
+    uint64_t threshold_total = 0;
+    for (uint64_t iter = 0; iter < kMaxIter; ++iter) {
+      v.assign(kSize, false);
+      for (;;) {
+        // size_t start = static_cast<size_t>(r.Next()) & kMask;
+        constexpr size_t kEvery = 207;
+        constexpr size_t kFactor = kSize / kEvery;
+        size_t start = static_cast<size_t>(r.Next() % kFactor) * kEvery;
+        size_t count;
+        if (approach == kRandom) {
+          count = 1;
+        } else {
+          // count = 85;
+          count = (static_cast<size_t>(r.Next()) & 255) + 85;
+        }
+        for (size_t i = 0; i < count; ++i) {
+          size_t pos;
+          if (approach == kAddition) {
+            pos = (start + i) & kMask;
+          } else {
+            pos = start ^ i;
+          }
+          if (v[pos]) {
+            goto next_iter;
```

Comment:
Can we do this test without a goto?  I have not seen a goto since Fortran and would rather keep it that way...